readthedocs/builds/models.py

```diff
@@ -200,6 +198,28 @@ def __str__(self):
     def is_external(self):
         return self.type == EXTERNAL
 
+    @property
+    def explicit_name(self):
+        """
+        Version name that is explicit about external origins.
+
+        For example, if a version originates from GitHub pull request #4, then
+        ``version.explicit_name == "#4 (PR)"``.
+
+        On the other hand, Versions associated with regular ReadTheDocs builds
+        (e.g. new tags or branches), simply return :obj:`~.verbose_name`.
+        This means that a regular git tag named **v4** will correspond to
+        ``version.explicit_name == "v4"``.
+        """
+        external_origin = external_version_name(self)
+
+        if not external_origin:
```

Comment:
We should check for `if not self.type == EXTERNAL` here or similar -- it's cleaner than depending on the side effects of another function.